src/http/api/account.ts

```diff
@@ -304,8 +305,34 @@ export function createGetAccountPostsHandler(
                 if (postResult instanceof Error) {
                     throw postResult;
                 }
-                result.results = postResult.results;
-                result.nextCursor = postResult.nextCursor;
+                if (isError(postResult)) {
+                    const error = getError(postResult);
+                    switch (error) {
+                        case 'invalid-handle':
+                            logger.error(`Invalid handle ${handle}`);
+                            return new Response(null, { status: 400 });
+                        case 'missing-default-account':
+                            logger.error('Missing default account');
+                            return new Response(null, { status: 400 });
+                        case 'invalid-next-parameter':
+                            logger.error('Invalid next parameter');
+                            return new Response(null, { status: 400 });
+                        case 'actor-not-found':
+                            logger.error(`Actor not found for ${handle}`);
+                            return new Response(null, { status: 400 });
+                        case 'error-getting-outbox':
+                            logger.error(`Error getting outbox for ${handle}`);
+                            return new Response(null, { status: 500 });
```

Comment:
Same here about 500 - we should never explicitly return a 500, they are for unexpected errors only. I think this should probably be an empty posts list too!